app/gui2/src/components/widgets/Dropdown.vue

```diff
@@ -0,0 +1,111 @@
+<script setup lang="ts">
+import SortAscendingIcon from '@/assets/icons/sort_ascending.svg'
+import SortDescendingIcon from '@/assets/icons/sort_descending.svg'
+import SortIcon from '@/assets/icons/sort.svg'
+import { computed, ref } from 'vue'
+
+enum SortDirection {
+  none = 'none',
+  ascending = 'ascending',
+  descending = 'descending',
+}
+
+const props = defineProps<{ color: string; selectedValue: string | null; values: string[] }>()
+const emit = defineEmits<{ click: [index: number] }>()
+
+const sortDirection = ref<SortDirection>(SortDirection.none)
+
+const sortedValuesAndIndices = computed(() => {
+  const valuesAndIndices = props.values.map<[value: string, index: number]>((value, index) => [
+    value,
+    index,
+  ])
+  switch (sortDirection.value) {
+    case SortDirection.none: {
+      return valuesAndIndices
+    }
+    case SortDirection.ascending: {
+      return valuesAndIndices.sort((a, b) => (a[0] > b[0] ? 1 : a[0] < b[0] ? -1 : 0))
+    }
+    case SortDirection.descending: {
+      return valuesAndIndices.sort((a, b) => (a[0] > b[0] ? -1 : a[0] < b[0] ? 1 : 0))
+    }
+  }
+})
+
+const ICON_LOOKUP: Record<SortDirection, string> = {
+  [SortDirection.none]: SortIcon,
+  [SortDirection.ascending]: SortAscendingIcon,
+  [SortDirection.descending]: SortDescendingIcon,
+}
+
+const NEXT_SORT_DIRECTION: Record<SortDirection, SortDirection> = {
+  [SortDirection.none]: SortDirection.ascending,
+  [SortDirection.ascending]: SortDirection.descending,
+  [SortDirection.descending]: SortDirection.none,
+}
+</script>
+
+<template>
+  <div :tabindex="-1" class="Dropdown" @wheel.stop>
+    <ul class="list" :style="{ background: color }">
+      <template v-for="[value, index] in sortedValuesAndIndices" :key="value">
+        <li v-if="value === selectedValue">
+          <div class="selected-item"><span v-text="value"></span></div>
+        </li>
+        <li v-else class="selectable-item button" @click="emit('click', index)">
+          <span v-text="value"></span>
+        </li>
+      </template>
+      <img
+        class="sort button"
+        :src="ICON_LOOKUP[sortDirection]"
+        @click="sortDirection = NEXT_SORT_DIRECTION[sortDirection]"
+      />
+    </ul>
+  </div>
+</template>
+
+<style scoped>
+.Dropdown {
+  position: absolute;
+  user-select: none;
+  overflow: auto;
+  clip-path: inset(0 round 8px);
+  width: min-content;
+  top: 100%;
+  margin-top: 4px;
+  height: 136px;
+  -ms-overflow-style: none;
+  scrollbar-width: none;
+}
+
+.Dropdown::-webkit-scrollbar {
+  display: none;
+}
+
+.list {
+  position: relative;
+  color: var(--color-text-light);
+  padding-left: 8px;
+  padding-right: 8px;
+  width: min-content;
+}
+
+.list > * {
+  position: relative;
```

Comment:
i usually add it because absolute/relative both create a stacking context, so when something has a position absolute sibling that overlaps it, then the text may disappear under the sibling. in this case it's probably not needed though,. I'll try removing it and report back

Reply:
Stacking contexts or other such constructs should be added as required, not "just in case". Try to keep the styles to minimum that is necessary at present.

Reply:
yeah i think i added it out of habit? note that this *is* needed for the toolbar and circular menu for visualizations, since those *do* have a 64px background blur.